src/harness/reference_models/fss_purge/fss_purge.py

```diff
@@ -0,0 +1,357 @@
+
+    oobe_interference = mcbsd + gcbsd - lcbsd + gfss
+    cbsd['oobe_interference'] = oobe_interference
+
+def getMcbsdValue(grant_request):
+  """ Identifies the MCBSD value for a grant.
+
+  MCBSD is the CBSD conducted power (in dBm) on the frequency segment, 
+  using the Tx mask of CBSD according to R0-DEV-05(e).
+
+  Args:
+    grant_request: A GrantData object.
+  Returns:
+    MCBSD value
+  """
+  if grant_request['operationParam']['operationFrequencyRange']['highFrequency'] > 3690000000:
+      mcbsd = -13
+  else:
+      mcbsd = -25
+
+  return mcbsd
+
+def getAntennaGainTowardsFss(incidence_angle, cbsd):
+  """Get antenna gain of CBSD in direction of FSS protection point [R2-SGN-20]
+
+  Args:
+    incidence_angle: Incidence angle as identified by ITM model.
+    cbsd: A CbsdData object
+  Returns
+    Antenna gain towards FSS in dBi.
+  """
+
+  cbsd_antenna_gain = antenna.GetStandardAntennaGains( incidence_angle.hor_cbsd,\
+                       cbsd['registrationRequest']['installationParam']['antennaAzimuth'],
+                       cbsd['registrationRequest']['installationParam']['antennaBeamwidth'],
+                       cbsd['registrationRequest']['installationParam']['antennaGain'])
+
+  return cbsd_antenna_gain
+
+def getMeanPathLossFromCbsdToFss(cbsd, fss):
+  """Get mean ITM path loss from CBSD to FSS protection point.
+
+  Args:
+    cbsd: A CbsdData object
+    fss_entity: The FssProtectionPoint tuple of the FSS.
+  Returns:
+    Mean path loss from CBSD to FSS in dBi.
+  """
+
+  # Extract the CBSD parameters
+  cbsd_latitude = cbsd['registrationRequest']['installationParam']['latitude']
+  cbsd_longitude = cbsd['registrationRequest']['installationParam']['longitude']
+  cbsd_height = cbsd['registrationRequest']['installationParam']['height']
+  cbsd_height_type = cbsd['registrationRequest']['installationParam']['heightType']
+  cbsd_indoor_dep = cbsd['registrationRequest']['installationParam']['indoorDeployment']
+  if cbsd_height_type == 'AMSL':
+      cbsd_height_agl = convertAmslToAgl(cbsd_latitude, cbsd_longitude, cbsd_height)
+  else:
+      cbsd_height_agl = cbsd_height
+
+  # Invoking ITM propagation model
+  db_loss, incidence_angle, _ = wf_itm.CalcItmPropagationLoss(
+                                         cbsd_latitude,
+                                         cbsd_longitude,
+                                         cbsd_height_agl,
+                                         fss.latitude,
+                                         fss.longitude,
+                                         fss.height_agl,
+                                         cbsd_indoor_dep,
+                                         reliability=-1,
+                                         freq_mhz=FREQ_PROP_MODEL)
+
+  return db_loss, incidence_angle
+
+def convertAmslToAgl(latitude, longitude, height_amsl):
+  """ Convert AMSL height to AGL height."""
+
+  altitude_cbsd = terrain.GetTerrainElevation(latitude, longitude)

```

Comment:
`terrain` is a module. You need to call the function on a terrain driver.
Currently there is only one terrain driver initialized in the wf_itm module. I should perhaps move it to the terrain.py module.
For now you can change into:
```
altitude_cbsd = wf_itm.terrainDriver.GetTerrainElevation(latitude, longitude)
```

Reply:
Corrected.